sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1090,6 +1090,17 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val CACHE_DISABLE_CONFIGS_ENABLED =
+    buildConf("spark.sql.cache.disableConfigs.enabled")
+      .internal()
+      .doc(s"When true, some configs are disabled during executing cache plan that is to avoid " +
+        s"performance regression if other queries hit the cached plan. Currently, the disabled " +
+        s"configs include: ${ADAPTIVE_EXECUTION_ENABLED.key} and " +
+        s"${AUTO_BUCKETED_SCAN_ENABLED.key}.")
```

Comment:
How about having a configuration to take a list that should be disabled for cached plans (by default both AQE and auto bucketed scan)?

Reply:
how about https://github.com/apache/spark/pull/32482#issuecomment-836259432 ?

Reply:
👌

Reply:
+1

Reply:
thank you all ! has refined the config name with `spark.sql.optimizer.canChangeCachedPlanOutputPartitioning`.